src/main/cli/commandLine.ts

```diff
@@ -0,0 +1,92 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { readerActions } from "readium-desktop/common/redux/actions";
+import { OpdsFeedRepository } from "readium-desktop/main/db/repository/opds";
+import { PublicationRepository } from "readium-desktop/main/db/repository/publication";
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { CatalogService } from "readium-desktop/main/services/catalog";
+import { Store } from "redux";
+import { URL } from "url";
+import { isArray } from "util";
+
+export async function cli_(filePath: string[] | string) {
+
+    let publicationOpenRequested = false;
+    let returnValue = true;
+    const filePathArray = isArray(filePath) ? filePath : [filePath];
+
+    for (const fp of filePathArray) {
+        // import and read publication
+        const catalogService = container.get("catalog-service") as CatalogService;
+        const publication = await catalogService.importFile(fp);
+        const store = container.get("store") as Store<RootState>;
+        if (publication) {
+            if (!publicationOpenRequested) {
+                store.dispatch({
+                    type: readerActions.ActionType.OpenRequest,
+                    payload: {
+                        publication: {
+                            identifier: publication.identifier,
+                        },
+                    },
+                });
+                publicationOpenRequested = true;
+            }
+        } else {
+            process.stderr.write(`Publication error for "${filePath}"\n`);
+            returnValue = false;
+        }
+    }
+    return returnValue;
+}
+
+export async function cliImport(filePath: string) {
+    try {
+        const catalogService = container.get("catalog-service") as CatalogService;
+        if (!await catalogService.importFile(filePath)) {
+            throw new Error();
```

Comment:
Is this `throw` statement meant to reject the `cliImport` promise, or is it intentional for the `catch` immediately below?
* If the former, then this does not work (the `cliImport` promise will resolve with `false` instead of the desired behaviour)
* If the latter, then I would suggest `return false` instead of needlessly throwing an exception.

Reply:
it's for return false when the catch appear or importFile returns null

Reply:
it's for return false when the catch appear or importFile returns null

Reply:
In that case `return false`. Throwing an exception is (1) unnecessary, and more importantly (2) misleading (because it implies that the intent is to reject the promise, when in fact it is to resolve it with `false`).

Reply:
done